compiler/shared_ast/expr.ml

```diff
@@ -702,18 +698,27 @@ let rec free_vars : ('a, 't) gexpr -> ('a, 't) gexpr Var.Set.t = function
 
 let rec skip_wrappers : type a. (a, 'm) gexpr -> (a, 'm) gexpr = function
   | EApp { f = EOp { op = Log _; _ }, _; args = [e] }, _ -> skip_wrappers e
+  | EApp { f = EApp { f = EOp { op = Log _; _ }, _; args = [f] }, _; args }, m
+    ->
+    skip_wrappers (EApp { f; args }, m)
   | EErrorOnEmpty e, _ -> skip_wrappers e
+  | EDefault { excepts = []; just = ELit (LBool true), _; cons = e }, _ ->
+    skip_wrappers e
```

Comment:
I would put that as an optimization in `shared_ast/optimizations.ml` rather than just a `skip_wrapper` thing.

Reply:
It's useful besides optimisations: typically, you want to use it before matching on some constructs to ignore intermediate wrappers. Or sometimes when you want a concise printing of an expression.

Note also that it's shallow, as mentionned in the mli: it will remove any number of nested wrappers, but only "on top" of the expression.